metrics/sinks/influxdb/influxdb.go

```diff
@@ -257,17 +257,48 @@ func (sink *influxdbSink) createDatabase() error {
 	if sink.dbExists {
 		return nil
 	}
+
+	//check whether the specific database exist or not
 	q := influxdb.Query{
-		Command: fmt.Sprintf(`CREATE DATABASE %s WITH NAME "default"`, sink.c.DbName),
+		Command: "SHOW DATABASES",
 	}
 
-	if resp, err := sink.client.Query(q); err != nil {
-		if !(resp != nil && resp.Err != nil && strings.Contains(resp.Err.Error(), "already exists")) {
-			err := sink.createRetentionPolicy()
-			if err != nil {
-				return err
+	resp, err := sink.client.Query(q)
+
+	if err != nil {
+		return err
+	}
+
+	if resp != nil && len(resp.Results) == 1 && len(resp.Results[0].Series) == 1 {
+		values := resp.Results[0].Series[0].Values
+		for _, name := range values {
+			if sink.c.DbName == name[0].(string) {
+				sink.dbExists = true
+				//If the database is exist,we should exit right now.
+				return nil
 			}
 		}
+
+		q = influxdb.Query{
+			Command: fmt.Sprintf(`CREATE DATABASE %s`, sink.c.DbName),
+		}
+
+		_, err := sink.client.Query(q)
+		if err != nil {
+			return err
+		}
+
+		/**
+		heapster should on create retention policy on the creation of database.
+		although you can change the retention duration in params,
+		but heapster will not alter a existing database's retention policies.
+		*/
+		err = sink.createRetentionPolicy()
+		if err != nil {
+			return err
```

Comment:
Also add an error log here to clearly recording the specific error for create retention policy.